style/checkstyle.xml

```diff
@@ -270,7 +270,7 @@
         <module name="IllegalImport">
             <property name="regexp" value="true"/>
             <property name="illegalPkgs"
-                      value="org\.apache\.commons, com\.google\.common, org\.apache\.log4j, org\.codehaus\.jackson"/>
+                      value="org\.apache\.commons, org\.apache\.log4j, org\.codehaus\.jackson"/>
```

Comment:
It looks like the guava cache is forbidden, we can switch to caffeine cache.

Reply:
> It looks like the guava cache is forbidden, we can switch to caffeine cache.

I tried to replace guava with caffeine, but I found that the `caffeine::Interner` class is available only in versions 3.0 and above, but 3.x does not support java8. So maybe we should keep importing guava.

Reply:
you can use `caffeine::cache` instead.

Reply:
> you can use `caffeine::cache` instead.

done

Reply:
I see the `Interner` is still used.